src/modules/Dashboard/features/Applet/Schedule/ActivityForm/ActivityForm.tsx

```diff
@@ -1,67 +1,189 @@
-      if (dirtyFields.availability) {
-        if (getValues().availability) {
-          setRemoveAllEventsPopupVisible(true);
+    const activityOrFlowId = watch('activityOrFlowId');
+    const alwaysAvailable = watch('alwaysAvailable');
+    const oneTimeCompletion = watch('oneTimeCompletion');
+    const timerType = watch('timerType');
+    const timerDuration = watch('timerDuration');
+    const idleTime = watch('idleTime');
+    const periodicity = watch('periodicity');
+    const startTime = watch('startTime');
+    const endTime = watch('endTime');
+    const accessBeforeSchedule = watch('accessBeforeSchedule');
+    const date = watch('date');
+    const startEndingDate = watch('startEndingDate');
+
+    // TODO: add get events callback
+    const { execute, error } = useAsync(createEventApi);
+
+    const handleCreateEvent = async () => {
+      if (!appletId) {
+        return;
+      }
+
+      const regexWithFlow = /^flow-/;
+      const isFlowId = regexWithFlow.test(activityOrFlowId);
+
+      const body: CreateEventType['body'] = {
+        timerType,
+        ...(isFlowId
+          ? { flowId: activityOrFlowId.replace(regexWithFlow, '') }
+          : { activityId: activityOrFlowId }),
+      };
+
+      if (timerType === TimerType.Timer) {
+        body.timer = addSecondsToHourMinutes(timerDuration);
+      }
+
+      if (timerType === TimerType.Idle) {
+        body.timer = addSecondsToHourMinutes(idleTime);
+      }
+
+      if (alwaysAvailable) {
+        body.oneTimeCompletion = oneTimeCompletion;
+        body.periodicity = {
+          type: 'ALWAYS',
+          selectedDate: format(defaultStartDate as Date, DateFormats.YearMonthDay),
+        };
+      } else {
+        body.startTime = addSecondsToHourMinutes(startTime);
+        body.endTime = addSecondsToHourMinutes(endTime);
+        body.accessBeforeSchedule = accessBeforeSchedule;
+
+        body.periodicity = {
+          type: periodicity,
+          ...(periodicity === Periodicity.Once
+            ? {
+                selectedDate: convertDateToYearMonthDay(date as Date),
+              }
+            : {
+                startDate: convertDateToYearMonthDay(startEndingDate[0] as Date),
+                ...(startEndingDate[1] && {
+                  endDate: convertDateToYearMonthDay(startEndingDate[1] as Date),
+                }),
+              }),
+          ...(defaultStartDate &&
+            (periodicity === Periodicity.Weekly || periodicity === Periodicity.Monthly) && {
+              selectedDate: format(defaultStartDate as Date, DateFormats.YearMonthDay),
+            }),
+        };
+      }
+
+      // TODO: add individual schedule events create with the respondentId when the API is ready and is connected
+      // TODO: add Notifications and Reminders when the API is ready, add time selected error due to the story
+      await execute({ appletId, body });
+    };
+
+    const submitForm = async () => {
+      if (dirtyFields.alwaysAvailable) {
```

Comment:
please use early return to avoid nesting

Reply:
done